core/src/main/scala/kafka/server/ControllerServer.scala

```diff
@@ -173,8 +174,12 @@ class ControllerServer(
         setMetrics(new QuorumControllerMetrics(KafkaYammerMetrics.defaultRegistry())).
         setCreateTopicPolicy(createTopicPolicy.asJava).
         setAlterConfigPolicy(alterConfigPolicy.asJava).
-        setConfigurationValidator(new ControllerConfigurationValidator()).
-        build()
+        setConfigurationValidator(new ControllerConfigurationValidator())
+      authorizer match {
```

Comment:
It's not that simple because you could have an authorizer which is not a ClusterAuthorizer, and therefore should not be passed to the Controller.

The Controller never authorizes things (since that's a blocking operation) but it does create and remove acls based on records, which is why it needs the object. I wish we had two separate public APIs instead of one: one to change ACLs and one to actually authorize based on the current ACLs. But it's probably not worth refactoring it to be like that now.

Reply:
Makes sense.